android/src/main/java/nordpol/ui/NfcGuideView.java

```diff
@@ -0,0 +1,276 @@
+ * used.
+ *
+ * Usage: Declare the View and call the different states directly on it. The
+ * available states are declared in the enum NfcGuideViewStatus and are called
+ * with the method setCurrentStatus. The NFC device graphic shown to the user
+ * can be changed. This is currently only possible with the XML attribute
+ * nfc_device. The "graphic-states" available for the attribute are declared in
+ * the android/src/main/res/values/attrs.xml file.
+ */
+public class NfcGuideView extends RelativeLayout {
+    private static final int GUIDE_ITEMS_START_DISTANCE = 100;
+    private static final int ANIMATION_DURATION_SHORT = 100;
+    private static final int ANIMATION_DURATION_MEDIUM = 200;
+    private static final int VALUE_ANIMATOR_DEFAULT_START_DELAY = 0;
+    private final int START_DELAY_LONG = 500;
+    private int mNfcGuidePhoneWidth = 0;
+    private int mNfcGuideHandWidth = 0;
+    private int mRootViewWidth = 0;
+
+    private View mRootView;
+    private ImageView mNfcGuidePhone;
+    private ImageView mNfcGuideHand;
+    private ProgressBar mProgressBar;
+    private ImageView mStatusPositive;
+    private ImageView mStatusNegative;
+
+    public enum NfcGuideViewStatus {
+        STARTING_POSITION,
+        TRANSFERRING,
+        DONE,
+        FAIL
+    }
+
+    public NfcGuideView(Context context, AttributeSet attrs) {
+        super(context, attrs);
+
+        TypedArray typedArray = context.getTheme().obtainStyledAttributes(
+          attrs,
+          R.styleable.NfcGuideView,
+          0, 0);
+
+        int nfcDeviceDesign = 0;
+
+        try {
+          nfcDeviceDesign = typedArray.getInteger(R.styleable.NfcGuideView_nfc_device, 0);
+        } finally {
+          typedArray.recycle();
+        }
+
+        LayoutInflater mLayoutInflator = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+        mRootView = mLayoutInflator.inflate(R.layout.nfc_guide_view, this, true);
+
+        mProgressBar = (ProgressBar) mRootView.findViewById(R.id.nfc_guide_view_progress_bar);
+        mStatusPositive = (ImageView) mRootView.findViewById(R.id.nfc_guide_view_status_positive);
+        mStatusNegative = (ImageView) mRootView.findViewById(R.id.nfc_guide_view_status_negative);
+        mNfcGuidePhone = (ImageView) mRootView.findViewById(R.id.nfc_guide_view_phone);
+        mNfcGuideHand = (ImageView) mRootView.findViewById(R.id.nfc_guide_view_hand);
+
+        switch(nfcDeviceDesign) {
+        case 0:
+            mNfcGuideHand.setImageDrawable(getResDrawable(R.drawable.nfc_guide_view_hand_holding_card_fidesmo));
+            break;
+        case 1:
+            mNfcGuideHand.setImageDrawable(getResDrawable(R.drawable.nfc_guide_view_hand_holding_card_neutral));
+            break;
+        }
+
+        mRootView.getViewTreeObserver().addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
+            @SuppressWarnings("deprecation")
```

Comment:
@Franzaine Is it really good practice to add new code that uses deprecated functionality?